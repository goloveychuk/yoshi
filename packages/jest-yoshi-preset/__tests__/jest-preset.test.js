const { overrideJestConfig } = require('../jest-preset');

let logSpy;

beforeEach(() => {
  logSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
});

afterEach(() => {
  logSpy.mockRestore();
});

it('should not log in case of whitelist keys', () => {
  overrideJestConfig({ globals: 'foo' }, 'e2e');
  expect(logSpy).not.toHaveBeenCalled();
});

it('should log in case of non whitelist keys', () => {
  overrideJestConfig({ globals1: 'foo' }, 'e2e');

  expect(logSpy).toHaveBeenCalled();
  expect(logSpy.mock.calls[0][0]).toMatchSnapshot();
});
