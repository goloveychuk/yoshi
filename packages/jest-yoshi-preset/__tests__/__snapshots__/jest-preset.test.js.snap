// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should log in case of non whitelist keys 1`] = `
"[31m[39m
[31mYoshi only supports overriding these Jest options:[39m
[31m[39m
[31m[1m  â€¢ globals[22m.[39m
[31m[39m
[31mThese options in your jest-yoshi.config.js file are not currently supported by Yoshi:[39m
[31m[39m
[31m[1m  â€¢ e2e.globals1[22m[39m"
`;
